src/node_file.cc

```diff
@@ -639,37 +677,52 @@ static void Link(const FunctionCallbackInfo<Value>& args) {
     return TYPE_ERROR("src path required");
   if (len < 2)
     return TYPE_ERROR("dest path required");
-  if (!args[0]->IsString())
-    return TYPE_ERROR("src path must be a string");
-  if (!args[1]->IsString())
-    return TYPE_ERROR("dest path must be a string");
 
-  node::Utf8Value orig_path(env->isolate(), args[0]);
-  node::Utf8Value new_path(env->isolate(), args[1]);
+  BufferValue src(env->isolate(), args[0]);
+  ASSERT_PATH(src)
+
+  BufferValue dest(env->isolate(), args[1]);
+  ASSERT_PATH(dest)
 
   if (args[2]->IsObject()) {
-    ASYNC_DEST_CALL(link, args[2], *new_path, *orig_path, *new_path)
+    ASYNC_DEST_CALL(link, args[2], *dest, UTF8, *src, *dest)
   } else {
-    SYNC_DEST_CALL(link, *orig_path, *new_path, *orig_path, *new_path)
+    SYNC_DEST_CALL(link, *src, *dest, *src, *dest)
   }
 }
 
 static void ReadLink(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
 
-  if (args.Length() < 1)
+  int len = args.Length();
+
+  if (len < 1)
     return TYPE_ERROR("path required");
-  if (!args[0]->IsString())
-    return TYPE_ERROR("path must be a string");
 
-  node::Utf8Value path(env->isolate(), args[0]);
+  BufferValue path(env->isolate(), args[0]);
+  ASSERT_PATH(path)
 
-  if (args[1]->IsObject()) {
-    ASYNC_CALL(readlink, args[1], *path)
+  enum encoding encoding = len > 1 && args[1]->IsString() ?
+      ParseEncoding(env->isolate(), args[1], UTF8) : UTF8;
```

Comment:
You can write this as `enum encoding encoding = ParseEncoding(env->isolate(), args[1], UTF8);` - ParseEncoding() returns the default encoding when second_arg->IsString() == false.